pkg/azure/api/api.go

```diff
@@ -386,6 +410,34 @@ func (c *Client) GetInstances(ctx context.Context, subnets ipamTypes.SubnetMap)
 	return instances, nil
 }
 
+var vmssRe = regexp.MustCompile(`[Vv]irtual[Mm]achine[Ss]cale[Ss]ets\/(.*)\/[Vv]irtual[Mm]achines\/(.*)`)
+
+// GetInstance returns the interfaces of a given instance
+func (c *Client) GetInstance(ctx context.Context, subnets ipamTypes.SubnetMap, instanceID string) (*ipamTypes.Instance, error) {
+	instance := ipamTypes.Instance{}
+	instance.Interfaces = map[string]ipamTypes.InterfaceRevision{}
+
+	parts := vmssRe.FindStringSubmatch(instanceID)
+	if len(parts) != 3 {
+		return nil, fmt.Errorf("instance ID %q does not match expected format", instanceID)
+	}
+	virtualMachineScaleSetName := parts[1]
+	virtualmachineIndex := parts[2]
+
+	networkInterfaces, err := c.listVirtualMachineScaleSetVMNetworkInterfaces(ctx, virtualMachineScaleSetName, virtualmachineIndex)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, networkInterface := range networkInterfaces {
+		if id, azureInterface := parseInterface(networkInterface, subnets, c.usePrimary); id != "" {
```

Comment:
Little confused here: `parseInterface` returns the instance ID https://github.com/cilium/cilium/blob/165c7e6a9a358ab65f78c40622094c96b7704a18/pkg/azure/api/api.go#L304-L308 but here we're treating it like an interface ID by setting it in `Interfaces[id]`. Should we be using `azureInterface.InterfaceID()` instead?

Reply:
That's a very good catch, thanks a lot, fixed it.